plugins/catalog-backend-module-bitbucket-cloud/src/BitbucketCloudEntityProvider.ts

```diff
@@ -174,7 +194,108 @@ export class BitbucketCloudEntityProvider implements EntityProvider {
     );
   }
 
-  private async findCatalogFiles(): Promise<string[]> {
+  /** {@inheritdoc @backstage/plugin-events-node#EventSubscriber.supportsEventTopics} */
+  supportsEventTopics(): string[] {
+    return [BitbucketCloudEntityProvider.TOPIC];
+  }
+
+  /** {@inheritdoc @backstage/plugin-events-node#EventSubscriber.onEvent} */
+  async onEvent(topic: string, payload: unknown): Promise<void> {
+    if (topic !== BitbucketCloudEntityProvider.TOPIC) {
+      return;
+    }
+
+    if ('push' in (payload as object)) {
+      await this.onRepoPush(payload as Events.RepoPushEvent);
+    }
+  }
+
+  async onRepoPush(event: Events.RepoPushEvent): Promise<void> {
+    if (!this.catalogApi || !this.tokenManager) {
+      throw new Error(
+        `${this.getProviderName()} not well configured to handle repo:push. Missing CatalogApi and/or TokenManager.`,
```

Comment:
Not sure. This a configuration error and it would be great to fail fast at startup even, but unfortunately, it is not really clear until runtime / upon receiving an event. Therefore, I think it is fine to let the subscriber fail to handle the event. A warn log once wouldn't be good enough I think.

Reply:
Another option I could see is that we could have a second entity provider implementation which extends the existing one and extends it with events support. (e.g., `BitbucketCloudEntityProvider` and `BitbucketCloudEventBasedEntityProvider`).

This way, all those are not optional anymore. Similar to the `events` dependency at the module (we would have two modules).

Not sure if this is easier though. What do you think @freben ?

Reply:
hmm.. seems like keeping the api-report clean will be a bit tricky then.

Reply:
btw: I think this is comparable to
```
    if (!this.connection) {
      throw new Error('Not initialized');
    }
```

Reply:
another option could be to make these two required instead of optional. This would be a breaking change though

Reply:
I've changed it to throwing the error once